zfs should build  without  "---without-python or --disable-pyzfs"

try to build an "ARCH iso" 

how ever this is zfs-2.1.4  mkdksm conf has been to be patched:

==> dkms install --no-depmod zfs/2.1.4 -k 5.17.5-arch1-1
Deprecated feature: REMAKE_INITRD
configure: error: "Python >= 3.4 is required"

patch has been changed to (but still no luck):
--- scripts/dkms.mkconf.bk	2022-04-30 17:38:29.376581217 +0200
+++ scripts/dkms.mkconf	2022-04-30 17:39:16.753437330 +0200
@@ -28,14 +28,9 @@
 PRE_BUILD="configure
   --prefix=/usr
   --with-config=kernel
-  --with-linux=\$(
-    if [ -e "\${kernel_source_dir/%build/source}" ]
-    then
-      echo "\${kernel_source_dir/%build/source}"
-    else
-      echo "\${kernel_source_dir}"
-    fi
-  )
+  --without-python --disable-pyzfs
+  --disable-pyzfs
+  --with-linux="\${kernel_source_dir}"
   --with-linux-obj="\${kernel_source_dir}"
   \$(
     [[ -n \"\${ICP_ROOT}\" ]] && \\
@@ -69,7 +63,7 @@
 "
 AUTOINSTALL="yes"
 REMAKE_INITRD="no"
-MAKE[0]="make"
+MAKE[0]="make -C module/"
 STRIP[0]="\$(
   [[ -r \${PACKAGE_CONFIG} ]] \\
   && source \${PACKAGE_CONFIG} \\

--- a/config/always-pyzfs.m4
+++ b/config/always-pyzfs.m4
@@ -47,6 +47,16 @@ AC_DEFUN([ZFS_AC_CONFIG_ALWAYS_PYZFS], [
 	AC_SUBST(DEFINE_PYZFS)
 
 	dnl #
+	dnl # Autodetection disables pyzfs if kernel or srpm config
+	dnl #
+	AS_IF([test "x$enable_pyzfs" = xcheck], [
+		AS_IF([test "x$ZFS_CONFIG" = xkernel -o "x$ZFS_CONFIG" = xsrpm ], [
+				enable_pyzfs=no
+				AC_MSG_NOTICE([Disabling pyzfs for kernel/srpm config])
+		])
+	])
+
+	dnl #
 	dnl # Python "packaging" (or, failing that, "distlib") module is required to build and install pyzfs
 	dnl #
 	AS_IF([test "x$enable_pyzfs" = xcheck -o "x$enable_pyzfs" = xyes], [

